src/heap_utils.cc

```diff
@@ -231,19 +232,103 @@ class BufferOutputStream : public v8::OutputStream {
   std::unique_ptr<JSString> buffer_;
 };
 
+namespace {
+static const int kMaxPath = 4096;
+
+class FileOutputStream : public v8::OutputStream {
+ public:
+  explicit FileOutputStream(FILE* stream) : stream_(stream) {}
+
+  virtual int GetChunkSize() {
+    return 65536;  // big chunks == faster
+  }
+
+  virtual void EndOfStream() {}
+
+  virtual WriteResult WriteAsciiChunk(char* data, int size) {
+    const size_t len = static_cast<size_t>(size);
+    size_t off = 0;
+
+    while (off < len && !feof(stream_) && !ferror(stream_))
+      off += fwrite(data + off, 1, len - off, stream_);
```

Comment:
Apologize for my ignorance, but are these calls to `fwrite`, `fopen`, `fclose` cross platform? No need for uv calls here instead?

Reply:
These *should* be safe and not using the libuv calls here should be fine. Writing the heapdump must be done synchronously so there's really not a lot of value in using the uv methods. If anyone feels strongly about it, then we can refactor a bit.

Reply:
@bnoordhuis Approving this PR is a good vote of confidence over my ignorance ;) LGTM.

Reply:
This should be okay, although it’s indeed unusual for our code base not to use the libuv API.